src/main/java/fr/sncf/osrd/infra_state/TVDSectionState.java

```diff
@@ -52,26 +57,37 @@ public void free(Simulation sim) throws SimulationError {
         sim.infraState.towerState.notifyFreed(sim, this.tvdSection);
     }
 
-    /** 
+    /**
      * Create an event to notify route that the tvd section is occupied
-     * 
+     *
      * @param sim the current simulation
      */
     public void occupy(Simulation sim) throws SimulationError {
-        assert isReserved();
+        if (isOccupied)
+            throw new SimulationError("TVD section we try to occupy is already occupied");
+        isOccupied = true;
+        // We need to notify the passive route first, as it may put the controlled routes in the CONFLICT state
+        callbackAllRoutes(sim, false);
+        callbackAllRoutes(sim, true);
+    }
+
+    /** Notifies all the passive or controlled routes that go through this TVD section */
+    private void callbackAllRoutes(Simulation sim, boolean controlled) throws SimulationError {
         for (var route : tvdSection.routeSubscribers) {
-            var routeState = sim.infraState.getRouteState(route.index);
-            routeState.onTvdSectionOccupied(sim);
+            if (route.isControlled == controlled) {
+                var routeState = sim.infraState.getRouteState(route.index);
+                routeState.onTvdSectionOccupied(sim, tvdSection);
+            }
         }
     }
 
-    /** 
+    /**
      * Create an event to notify route that the tvd section is not occupied anymore
-     * 
+     *
      * @param sim the current simulation
      */
     public void unoccupy(Simulation sim) throws SimulationError {
-        assert isReserved();
+        isOccupied = false;
```

Comment:
This should be done in a change

Reply:
I moved this in a change, but I believe we could try to factorize all the "single variable change" classes. There's a lot of boilerplate. Not in this branch but later on.